View/LoginScreen.xaml.cs

```diff
@@ -71,6 +72,16 @@ private void Login_Button_Click(object sender, RoutedEventArgs e)
                     this.Error.Content = "Password incorrect";
                     break;
                 }
+                case LoginResults.UserNotActive:
+                {
+                    var token = Guid.NewGuid().ToString();
+                    var userEmail = controller.GetUserFromEmailOrUsername(emailOrUsername).Email;

```

Comment:
Weet je zeker dat de email een gebruiker teruggeeft?
Anders kan dit null opleveren namelijk

Reply:
Uhm dan zou toch eerst langs de andere foutcode van user not found moeten komen

Reply:
Ja, maar die vang je toch niet af?

Reply:
Dus je zult of iets van een null check moeten doen, of een trycatch

Reply:
er staat een null check op lijn 37 van usercontroller waar die hem zou af moeten vangen

Reply:
Dat gebeurd alleen als je een user inlogd, ofterwijl UserController.Login aanroept.
Niet als je alleen GetUserFromEmailOrUsername gebruikt.

Reply:
Ja maar, je mag niet .Email doen op iets wat null kan zijn. 
Dus eerst checken of controller.GetUserFromEmailOrUsername(emailOrUsername) null is, zo niet dan kun je email ophalen